src/csharp/Grpc.Core.Tests/CallCancellationTest.cs

```diff
@@ -179,6 +179,122 @@ public async Task ServerStreamingCall_CancelClientSideRead()
             }
         }
 
+        [Test]
+        public async Task ServerStreamingCall_DisposeCallBeforeAllRead()
+        {
+            // Test to reproduce https://github.com/grpc/grpc/issues/8451
+            // "Client channel socket leaks unless read stream drained explicitly"
+            // This version of the test disposes the call before the response stream is
+            // fully read.
+
+            helper.ServerStreamingHandler = new ServerStreamingServerMethod<string, string>(async (request, responseStream, context) => {
+                try
+                {
+                    // send responses to the client with a short delay between each response
+                    foreach (string resp in request.Split(new[] { ' ' }))
+                    {
+                        await responseStream.WriteAsync(resp);
+                        await Task.Delay(50);
+                    }
+                    context.ResponseTrailers.Add("xyz", "");
+                }
+                catch (RpcException ex)
+                {
+                    // server may get an exception when the client cancels
+                    Assert.AreEqual(StatusCode.Cancelled, ex.Status.StatusCode);
```

Comment:
Checking the status code in the server side handler isn't necessary (you can just eat the exception or simply let the handler throw, the grpc glue code will handle that. 
Also, the status  code it's really a thing on the server side (as the server sends status code to the client, not the other way round).
So let's remove the try catch block?